### PR TITLE
For #6413: Adds more snackbar positioning logic

### PR DESCRIPTION
**Pull request by sblatz**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/account/TurnOnSyncFragment.kt [hunk 1]
```diff
@@ -17,15 +17,18 @@ import mozilla.components.concept.sync.AccountObserver
...
+import org.mozilla.fenix.components.FenixSnackbar
 import org.mozilla.fenix.components.metrics.Event
 import org.mozilla.fenix.ext.requireComponents
 import org.mozilla.fenix.ext.showToolbar
 
 @SuppressWarnings("TooManyFunctions")
 class TurnOnSyncFragment : Fragment(), AccountObserver {
 
+    private val safeArguments get() = requireNotNull(arguments)
+    private val args get() = TurnOnSyncFragmentArgs.fromBundle(safeArguments)
```
[the remark this answers is not in the document]
Comment (Baron-Severin) [reply app/src/main/java/org/mozilla/fenix/settings/account/TurnOnSyncFragment.kt]: Bump

File: app/src/main/java/org/mozilla/fenix/components/FenixSnackbar.kt [hunk 2]
```diff
@@ -26,7 +26,7 @@ class FenixSnackbar private constructor(
     parent: ViewGroup,
     content: View,
     contentViewCallback: FenixSnackbarCallback,
-    isError: Boolean
+    var isError: Boolean
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/fenix/components/FenixSnackbar.kt]: I don't think we're using this, was it added by mistake?

File: app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt [hunk 3]
```diff
@@ -87,7 +87,7 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), UserInteractionHan
             bookmarksController = DefaultBookmarkController(
                 context = context!!,
                 navController = findNavController(),
-                snackbarPresenter = BrowserSnackbarPresenter(view),
+                snackbar = FenixSnackbar.make(view, FenixSnackbar.LENGTH_LONG),
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt]: Shouldn't this one be `makeWithToolbarPadding`?

Comment (sblatz) [reply app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt]: No, this one appears in settings. The `EditBookmarkFragment` appears properly with padding on Browser :)

**Pull request merged**